src/components/viewer.ts

```diff
@@ -97,13 +99,19 @@ export class Viewer {
     async open(pdfFile: string, mode?: string): Promise<void> {
         const configuration = vscode.workspace.getConfiguration('latex-workshop')
         const tabEditorGroup = configuration.get('view.pdf.tab.editorGroup') as string
-        const viewer = mode ?? configuration.get<'tab' | 'browser' | 'singleton' | 'external'>('view.pdf.viewer', 'tab')
+        const viewerLocation = configuration.get<ViewerLocation>('view.pdf.viewer', 'tab')
+        if(mode === 'tab' && (viewerLocation === 'customEditor' || viewerLocation === 'singleton')) {
```

Comment:
Instead of using two variables here, `mode` and `viewerLocation` should be unified. This only involves changing the function at [here](https://github.com/James-Yu/LaTeX-Workshop/blob/40a065513c41972327a0c5d757253a3df4fb48a0/src/commander.ts#L82)